dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -64,14 +64,25 @@ inline void ParseSampleDesc(std::vector<SampleDesc>& samples_container,
   // Getting the components data
   std::string components_metadata;
   std::getline(index_file, components_metadata);
-  std::stringstream extensions_stream(components_metadata);
+  std::stringstream components_stream(components_metadata);
 
   // Reading consecutive components
   ComponentDesc component;
-  while (extensions_stream >> component.ext) {
-    DALI_ENFORCE(extensions_stream >> component.offset >> component.size,
-                 IndexFileErrMsg(index_path, line,
-                                 "size or offset corresponding to the extension not found"));
+  while (components_stream >> component.ext) {
+    if (index_version == "v1.2") {
```

Comment:
It will do for now, but in the long run, I think we should parse the version into a number - 1.2 would become 1002 or sth along these lines. For once, a string <major>.<minor> will only hold 7 more minor versions (up to 1.9) until non-equality comparison fails. Another issue is the (possibly) multiple string comparison - with numerical version we could just use `switch`.

Reply:
Added a follow up task for it. This would be too big change for this PR.